applications/compareSummary.cpp

```diff
@@ -0,0 +1,461 @@
+/*
+  Copyright 2016 Statoil ASA.
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <ert/ecl/ecl_sum.h>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <ert/util/stringlist.h>
+#include <ert/util/int_vector.h>
+#include <ert/util/bool_vector.h>
+#include <opm/common/ErrorMacros.hpp>
+void printHelp();
+
+
+struct Deviation {
+  double relative_deviation; 
+  double absolute_deviation; 
+};
+
+
+class SummaryReader {
+private:
+  ecl_sum_type* ecl_sum1;
```

Comment:
I think we now can use the C++11 style inline default initializers:
```
    ecl_sum_type* ecl_sum1 = nullptr;
```
That cuts down on the boilerplate and on possibility for errors, too!